spec/features/reservations_spec.rb

```diff
@@ -545,4 +545,24 @@
       end
     end
   end
+  context 'redirects properly' do
+    before(:each) { empty_cart }
+    it 'handles direct url visit to /reservation/new with empty cart' do
+      visit root_path
+      visit new_reservation_path
+      expect(page.current_url).to include(root_path)
+      expect(page).to have_content('Catalog')
+    end
+    it 'handles press reserve button with empty cart' do
+      sign_in_as_user(@admin)
+      visit root_path
+      first('.equipment_title_link').click
+      expect(page.current_url).to include(equipment_models_path)
+      expect(page).to have_content('Description')
+      click_link('Reserve')
+      expect(page.current_url).to include(equipment_models_path)
+      expect(page).to have_content('Description')
+      sign_out
```

Comment:
If we're adding a `sign_in` call to the `before` block we should probably add `sign_out` to an `after` block for cleanliness.